src/main/scala/wdltool/graph/WomGraphPrint.scala

```diff
@@ -0,0 +1,130 @@
+      case other => throw new Exception(s"Can't read $filePath")
+    }
+  }
+
+  def generateWorkflowDigraph(file: String, auxFiles: Seq[String]): WorkflowDigraph = {
+
+    val womExecutable = if (file.toLowerCase().endsWith("wdl")) womExecutableFromWdl(file) else womExecutableFromCwl(file)
+    womExecutable.graph match {
+      case Valid(graph) =>
+        val digraph = listAllGraphNodes (graph)
+        WorkflowDigraph (dotSafe(womExecutable.entryPoint.name), digraph)
+      case Invalid(errors) =>
+        throw new IllegalArgumentException(errors.toList.mkString("\n", "\n", "\n"))
+    }
+  }
+
+  private val clusterCount: AtomicInteger = new AtomicInteger(0)
+
+  private def listAllGraphNodes(graph: Graph): NodesAndLinks = {
+
+    graph.nodes foldMap nodesAndLinks
+  }
+
+  private def nodesAndLinks(graphNode: GraphNode): NodesAndLinks = graphNode match {
+    case other => defaultNodesAndLinks(graphNode)
+  }
+
+  /**
+    * For most graph nodes, we draw a single box for the node, containing the name and input and output ports.
+    * For each input/output port, we include the links to upstream ports.
+    */
+  private def defaultNodesAndLinks(graphNode: GraphNode): NodesAndLinks = {
+    val clusterName = "cluster_" + clusterCount.getAndIncrement()
+    val portNodes = (graphNode.outputPorts ++ graphNode.inputPorts) map { p =>
+      s"${graphId(p)} [shape=${graphShape(p)} label=${graphName(p)}];"
+    }
+    val links = for {
+      inputPort <- graphNode.inputPorts
+      upstreamPort = inputPort.upstream
+    } yield graphId(upstreamPort)+" -> "+graphId(inputPort)
+
+    val singleNode =
+      s"""
+         |subgraph $clusterName {
+         |  style=filled;
+         |  fillcolor=${graphFillColor(graphNode)};
+         |  ${portNodes.mkString(sep="\n  ")}
+         |  ${graphId(graphNode)} [shape=plaintext label=${graphName(graphNode)}]
+         |}
+        """.stripMargin
+
+    NodesAndLinks(Set(singleNode), links)
+  }
+
+
+  private def dotSafe(s: String) = s""""${s.replaceAllLiterally("\"", "\\\"")}""""
+
+  private def graphId(g: GraphNode): String = dotSafe("NODE"+g.hashCode().toString)
+  private def graphId(p: GraphNodePort): String = dotSafe("PORT"+p.hashCode().toString)
+
+  private def graphName(p: GraphNodePort): String = dotSafe(p.name)
+  private def graphName(g: GraphNode): String = dotSafe(g match {
+    case c: CallNode =>
+      s"call ${c.name}"
+    case s: ScatterNode =>
+      s"scatter ${s.scatterVariableMapping.graphInputNode.name}"
+    case gin: GraphInputNode =>
+      s"${gin.womType.toWdlString} ${gin.name}"
+    case gon: GraphOutputNode =>
+      s"${gon.womType.toWdlString} ${gon.name}"
+    case other =>
+      throw new Exception(s"Can't name $other")
+  })
+
+  private def graphShape(p: GraphNodePort) = p match {
+    case _: InputPort => "oval"
+    case _: OutputPort => "hexagon"
+  }
+
+  private def graphFillColor(g: GraphNode) = g match {
```

Comment:
ToL personally I would have `implicit class`ed all these methods for `GraphNode` and `GraphNodePort` to help readability